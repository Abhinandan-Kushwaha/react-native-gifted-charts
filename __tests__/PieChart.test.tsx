/**
 * @format
 */

import 'react-native';
import React from 'react';
import ProgressPie from '../examples/PieChart/ProgressPie';
import SimplePie from '../examples/PieChart/SimplePie';
import SplitPie from '../examples/PieChart/SplitPie';
import ThreeDPie from '../examples/PieChart/ThreeDPie';
import PieSingleData from '../examples/PieChart/PieSingleData';
import PieWithZero from '../examples/PieChart/PieWithZero';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Pie and Donut chart tests', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });
  it('renders progress pie chart correctly', () => {
    const tree = renderer.create(<ProgressPie />).toJSON();
    jest.runAllTimers();
    expect(tree).toMatchSnapshot();
  });

  it('renders simple pie chart correctly', () => {
    const tree = renderer.create(<SimplePie />).toJSON();
    jest.runAllTimers();
    expect(tree).toMatchSnapshot();
  });

  it('renders split pie chart correctly', () => {
    const tree = renderer.create(<SplitPie />).toJSON();
    jest.runAllTimers();
    expect(tree).toMatchSnapshot();
  });

  it('renders 3D pie chart correctly', () => {
    const tree = renderer.create(<ThreeDPie />).toJSON();
    jest.runAllTimers();
    expect(tree).toMatchSnapshot();
  });

  it('renders pie chart correctly for Single data', () => {
    const tree = renderer.create(<PieSingleData />).toJSON();
    jest.runAllTimers();
    expect(tree).toMatchSnapshot();
  });

  it('renders pie chart with zero values correctly', () => {
    const tree = renderer.create(<PieWithZero />).toJSON();
    jest.runAllTimers();
    expect(tree).toMatchSnapshot();
  });

});
