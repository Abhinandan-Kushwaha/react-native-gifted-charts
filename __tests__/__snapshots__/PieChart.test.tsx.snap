// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pie and Donut chart tests renders 3D pie chart correctly 1`] = `
<View
  style={
    {
      "borderWidth": 1,
    }
  }
>
  <View
    style={
      {
        "height": 340,
        "overflow": "hidden",
        "width": 340,
      }
    }
  >
    <View
      style={
        {
          "position": "absolute",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          [
            {
              "backgroundColor": "transparent",
              "height": 390.99999999999994,
              "overflow": "hidden",
              "width": 340,
            },
            {
              "transform": [
                {
                  "rotateX": "55deg",
                },
              ],
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={340}
          bbWidth={340}
          focusable={false}
          height={340}
          meetOrSlice={0}
          minX={0}
          minY={0}
          pointerEvents="box-none"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 340,
                "width": 340,
              },
            ]
          }
          vbHeight={340}
          vbWidth={340}
          width={340}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGDefs>
              <RNSVGRadialGradient
                cx="50%"
                cy="50%"
                fx="50%"
                fy="50%"
                gradient={
                  [
                    0,
                    -1,
                    1,
                    -15238443,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="grad0"
                rx="50%"
                ry="50%"
              />
              <RNSVGRadialGradient
                cx="50%"
                cy="50%"
                fx="50%"
                fy="50%"
                gradient={
                  [
                    0,
                    -1,
                    1,
                    -8793378,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="grad1"
                rx="50%"
                ry="50%"
              />
              <RNSVGRadialGradient
                cx="50%"
                cy="50%"
                fx="50%"
                fy="50%"
                gradient={
                  [
                    0,
                    -1,
                    1,
                    -1218971,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="grad2"
                rx="50%"
                ry="50%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M 170 170 L 170 0 A 170 170 0 0 1 197.98108034772477 337.6814215784628 L 170 170"
              fill={
                {
                  "payload": 4279728853,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              strokeWidth={0}
            />
            <RNSVGPath
              d="M 170 170 L 197.98108034772477 337.6814215784628 A 170 170 0 0 1 18.31677943487292 93.239329087149 L 170 170"
              fill={
                {
                  "payload": 4286173918,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              strokeWidth={0}
            />
            <RNSVGPath
              d="M 170 170 L 18.31677943487292 93.239329087149 A 170 170 0 0 1 169.9999999999998 0 L 170 170"
              fill={
                {
                  "payload": 4293748325,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              strokeWidth={0}
            />
            <RNSVGCircle
              cx={297.0645228583504}
              cy={154.47113345227763}
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              r={26}
              responsible={true}
            />
            <RNSVGText
              content={null}
              dx={[]}
              dy={[]}
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontSize": 20,
                  "fontStyle": "normal",
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              responsible={true}
              rotate={[]}
              x={
                [
                  285.95341174723933,
                ]
              }
              y={
                [
                  159.47113345227763,
                ]
              }
            >
              <RNSVGTSpan
                content="54"
                dx={[]}
                dy={[]}
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={{}}
                rotate={[]}
                x={[]}
                y={[]}
              />
            </RNSVGText>
            <RNSVGCircle
              cx={67.26494902822532}
              cy={240.50999471478502}
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              r={26}
              responsible={true}
            />
            <RNSVGText
              content={null}
              dx={[]}
              dy={[]}
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontSize": 20,
                  "fontStyle": "normal",
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              responsible={true}
              rotate={[]}
              x={
                [
                  56.1538379171142,
                ]
              }
              y={
                [
                  245.50999471478502,
                ]
              }
            >
              <RNSVGTSpan
                content="40"
                dx={[]}
                dy={[]}
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={{}}
                rotate={[]}
                x={[]}
                y={[]}
              />
            </RNSVGText>
            <RNSVGCircle
              cx={103.23170346654555}
              cy={56.380275373113314}
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              r={26}
              responsible={true}
            />
            <RNSVGText
              content={null}
              dx={[]}
              dy={[]}
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontSize": 20,
                  "fontStyle": "normal",
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              responsible={true}
              rotate={[]}
              x={
                [
                  92.12059235543444,
                ]
              }
              y={
                [
                  61.380275373113314,
                ]
              }
            >
              <RNSVGTSpan
                content="20"
                dx={[]}
                dy={[]}
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={{}}
                rotate={[]}
                x={[]}
                y={[]}
              />
            </RNSVGText>
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          style={
            {
              "backgroundColor": "lightgray",
              "borderRadius": 170,
              "height": 340,
              "left": 0,
              "position": "absolute",
              "top": 34,
              "width": 340,
              "zIndex": -1,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderColor": "lightgray",
            "borderRadius": 68,
            "borderWidth": 6,
            "height": 136,
            "justifyContent": "center",
            "left": 92,
            "position": "absolute",
            "top": 112.49999999999997,
            "width": 136,
          },
          {
            "borderLeftWidth": 12,
            "borderTopWidth": 30,
            "transform": [
              {
                "rotateX": "55deg",
              },
            ],
          },
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "marginTop": 0,
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderColor": "lightgray",
            "borderRadius": 68,
            "borderWidth": 6,
            "height": 136,
            "justifyContent": "center",
            "left": 92,
            "position": "absolute",
            "top": 112.49999999999997,
            "width": 136,
          },
          {
            "borderLeftWidth": 12,
            "borderTopWidth": 30,
            "transform": [
              {
                "rotateX": "55deg",
              },
            ],
          },
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "marginTop": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Pie and Donut chart tests renders pie chart correctly for Single data 1`] = `
<View
  style={
    {
      "borderWidth": 1,
    }
  }
/>
`;

exports[`Pie and Donut chart tests renders pie chart with zero values correctly 1`] = `
<View
  style={
    {
      "borderWidth": 1,
    }
  }
>
  <View
    style={
      {
        "height": 240,
        "overflow": "hidden",
        "width": 240,
      }
    }
  >
    <View
      style={
        {
          "position": "absolute",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          [
            {
              "backgroundColor": "transparent",
              "height": 240,
              "overflow": "hidden",
              "width": 240,
            },
            undefined,
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={240}
          bbWidth={240}
          focusable={false}
          height={240}
          meetOrSlice={0}
          minX={0}
          minY={0}
          pointerEvents="box-none"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 240,
                "width": 240,
              },
            ]
          }
          vbHeight={240}
          vbWidth={240}
          width={240}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGDefs>
              <RNSVGRadialGradient
                cx="50%"
                cy="50%"
                fx="50%"
                fy="50%"
                gradient={
                  [
                    0,
                    -1,
                    1,
                    -16711681,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="grad0"
                rx="50%"
                ry="50%"
              />
              <RNSVGRadialGradient
                cx="50%"
                cy="50%"
                fx="50%"
                fy="50%"
                gradient={
                  [
                    0,
                    -1,
                    1,
                    -16744448,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="grad1"
                rx="50%"
                ry="50%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M 120 120 L 120 0 A 120 120 0 0 1 120.00471235952693 9.252638655254941e-8 L 120 120"
              fill={
                {
                  "payload": 4278255615,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              strokeWidth={0}
            />
            <RNSVGPath
              d="M 120 120 L 120.00471235952693 9.252638655254941e-8 A 120 120 0 1 1 119.99999999999997 0 L 120 120"
              fill={
                {
                  "payload": 4278222848,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              strokeWidth={0}
            />
            <RNSVGCircle
              cx={119.9982328651771}
              cy={205.99999998265133}
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              r={16}
              responsible={true}
            />
            <RNSVGText
              content={null}
              dx={[]}
              dy={[]}
              fill={
                {
                  "payload": 4286578816,
                  "type": 0,
                }
              }
              font={
                {
                  "fontSize": 16,
                  "fontStyle": "normal",
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              responsible={true}
              rotate={[]}
              x={
                [
                  111.10934397628822,
                ]
              }
              y={
                [
                  209.99999998265133,
                ]
              }
            >
              <RNSVGTSpan
                content="1"
                dx={[]}
                dy={[]}
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={{}}
                rotate={[]}
                x={[]}
                y={[]}
              />
            </RNSVGText>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderColor": "lightgray",
            "borderRadius": 48,
            "borderWidth": 0,
            "height": 96,
            "justifyContent": "center",
            "left": 72,
            "position": "absolute",
            "top": 72,
            "width": 96,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "marginTop": 0,
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderColor": "lightgray",
            "borderRadius": 48,
            "borderWidth": 0,
            "height": 96,
            "justifyContent": "center",
            "left": 72,
            "position": "absolute",
            "top": 72,
            "width": 96,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "marginTop": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Pie and Donut chart tests renders progress pie chart correctly 1`] = `
<View
  style={
    {
      "borderWidth": 1,
    }
  }
/>
`;

exports[`Pie and Donut chart tests renders simple pie chart correctly 1`] = `
<View
  style={
    {
      "borderWidth": 1,
    }
  }
>
  <View
    style={
      {
        "height": 240,
        "overflow": "hidden",
        "width": 240,
      }
    }
  >
    <View
      style={
        {
          "position": "absolute",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          [
            {
              "backgroundColor": "transparent",
              "height": 240,
              "overflow": "hidden",
              "width": 240,
            },
            undefined,
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={240}
          bbWidth={240}
          focusable={false}
          height={240}
          meetOrSlice={0}
          minX={0}
          minY={0}
          pointerEvents="box-none"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 240,
                "width": 240,
              },
            ]
          }
          vbHeight={240}
          vbWidth={240}
          width={240}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGDefs>
              <RNSVGRadialGradient
                cx="50%"
                cy="50%"
                fx="50%"
                fy="50%"
                gradient={
                  [
                    0,
                    -1,
                    1,
                    -15238443,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="grad0"
                rx="50%"
                ry="50%"
              />
              <RNSVGRadialGradient
                cx="50%"
                cy="50%"
                fx="50%"
                fy="50%"
                gradient={
                  [
                    0,
                    -1,
                    1,
                    -8793378,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="grad1"
                rx="50%"
                ry="50%"
              />
              <RNSVGRadialGradient
                cx="50%"
                cy="50%"
                fx="50%"
                fy="50%"
                gradient={
                  [
                    0,
                    -1,
                    1,
                    -1218971,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="grad2"
                rx="50%"
                ry="50%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M 120 120 L 120 0 A 120 120 0 0 1 126.85065730353217 239.80429247113625 L 120 120"
              fill={
                {
                  "payload": 4279728853,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              strokeWidth={0}
            />
            <RNSVGPath
              d="M 120 120 L 126.85065730353217 239.80429247113625 A 120 120 0 0 1 0.44019229647703995 109.73099898253919 L 120 120"
              fill={
                {
                  "payload": 4286173918,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              strokeWidth={0}
            />
            <RNSVGPath
              d="M 120 120 L 0.44019229647703995 109.73099898253919 A 120 120 0 0 1 119.99999999999997 0 L 120 120"
              fill={
                {
                  "payload": 4293748325,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              strokeWidth={0}
            />
            <RNSVGCircle
              cx={209.96329735453702}
              cy={114.42995542856733}
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              r={22}
              responsible={true}
            />
            <RNSVGText
              content={null}
              dx={[]}
              dy={[]}
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontSize": 12,
                  "fontStyle": "normal",
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              responsible={true}
              rotate={[]}
              x={
                [
                  203.29663068787036,
                ]
              }
              y={
                [
                  117.42995542856733,
                ]
              }
            >
              <RNSVGTSpan
                content="54%"
                dx={[]}
                dy={[]}
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={{}}
                rotate={[]}
                x={[]}
                y={[]}
              />
            </RNSVGText>
            <RNSVGCircle
              cx={55.45813757169329}
              cy={179.72438117897644}
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              r={22}
              responsible={true}
            />
            <RNSVGText
              content={null}
              dx={[]}
              dy={[]}
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontSize": 12,
                  "fontStyle": "normal",
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              responsible={true}
              rotate={[]}
              x={
                [
                  48.79147090502663,
                ]
              }
              y={
                [
                  182.72438117897644,
                ]
              }
            >
              <RNSVGTSpan
                content="30%"
                dx={[]}
                dy={[]}
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={{}}
                rotate={[]}
                x={[]}
                y={[]}
              />
            </RNSVGText>
            <RNSVGCircle
              cx={59.14425897533826}
              cy={50.69329759112355}
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              r={22}
              responsible={true}
            />
            <RNSVGText
              content={null}
              dx={[]}
              dy={[]}
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontSize": 12,
                  "fontStyle": "normal",
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                [
                  "fill",
                ]
              }
              responsible={true}
              rotate={[]}
              x={
                [
                  52.477592308671596,
                ]
              }
              y={
                [
                  53.69329759112355,
                ]
              }
            >
              <RNSVGTSpan
                content="26%"
                dx={[]}
                dy={[]}
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={{}}
                rotate={[]}
                x={[]}
                y={[]}
              />
            </RNSVGText>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderColor": "lightgray",
            "borderRadius": 70,
            "borderWidth": 0,
            "height": 140,
            "justifyContent": "center",
            "left": 50,
            "position": "absolute",
            "top": 50,
            "width": 140,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "marginTop": 0,
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderColor": "lightgray",
            "borderRadius": 70,
            "borderWidth": 0,
            "height": 140,
            "justifyContent": "center",
            "left": 50,
            "position": "absolute",
            "top": 50,
            "width": 140,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "marginTop": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Pie and Donut chart tests renders split pie chart correctly 1`] = `
<View
  style={
    {
      "borderWidth": 1,
    }
  }
/>
`;
